### PR TITLE
i.MX and sm License cleanup

### PR DESCRIPTION
**Pull request by MrVan**
Cleanup the license to only use the SPDX which is simplier.

### CODE REVIEW

**Comment by jforissier** [comment]
This is mostly OK except for the files where NXP is not the sole copyright owner (rules 2.2 and 2.3 in https://github.com/OP-TEE/optee_os/blob/master/documentation/copyright_and_license_headers.rst).
AFAIK, that's the rule adopted by the Linux kernel also (but feel free to correct me if I'm wrong).

What it means more precisely is:
- Changing files with only Freescale and/or NXP copyright is OK (rationale: Freescale has merged with NXP, and you're sending this PR on behalf of NXP)
- Files mentioning STMicroelectronics or Wind River should probably not be touched however.

**Comment by MrVan** [comment]
@jforissier Thanks for detailed information. Then I'll only cleanup the files containing Freescale and NXP.

File: core/arch/arm/plat-imx/a9_plat_init.S [hunk 1]
```diff
@@ -1,31 +1,7 @@
 /* SPDX-License-Identifier: BSD-2-Clause */
 /*
  * Copyright (c) 2014, STMicroelectronics International N.V.
- * All rights reserved.
```
Comment (etienne-lms) [inline core/arch/arm/plat-imx/a9_plat_init.S]: On behalf of STMicroelectronics, change approved.
etienne

Comment (jforissier) [reply core/arch/arm/plat-imx/a9_plat_init.S]: @etienne-lms fine by me. I'd like your `Acked-by:` added to the patch, then.

**Comment by etienne-lms** [comment]
`Ack-by: Etienne Carriere <etienne.carriere@st.com>`

**Comment by MrVan** [comment]
Updated with Tags applied and leave the windriver part as it is.

**Comment by etienne-lms** [comment]
@MrVan, you can add my `Acked-by: Etienne Carriere <etienne.carriere@linaro.org>` to the 2nd commit.

**Comment by MrVan** [comment]
Updated with Tags applied. Thanks.

**Comment by jforissier** [comment]
Thanks @MrVan, I have merged the commits manually because I have added by A-b, too.

**Pull request closed without merging**